File: content/search/source/includes/string-tutorial/code-snippets/nodejs/create-materialized-view.js
import { MongoClient } from 'mongodb';

async function main() {
    // Connect to MongoDB
    const uri = '<connectionString>';
    const client = new MongoClient(uri);

    // Define the aggregation pipeline
    const pipeline = [
        { $project: {
            lastScrapedDate: { 
              $dateToString: { 
                format: '%Y-%m-%d',
                date: '$last_scraped' 
              } 
            },
            propertyName: '$name',
            propertyType: '$property_type',
            accomodatesNumber: {
              $toString: '$accomodates',
              $toString: '$maximum_nights'
            },
            maximumNumberOfNights: {
              $toString: '$maximum_nights'
            }
          }
       },
       { $merge: {
            into: 'airbnb_mat_view',
            whenMatched: 'replace'
          }
       }
    ]; 


try {
        await client.connect();
        const database = client.db('sample_airbnb');
        const collection = database.collection('listingsAndReviews');

        // Run the aggregation
        const cursor = collection.aggregate(pipeline);

        while (await cursor.hasNext()) {
            // Consume the cursor to execute the pipeline
            await cursor.next();
        }
        console.log('Materialized view created!');

        // Explicitly close the cursor after processing
        await cursor.close();
        console.log('Aggregation cursor closed.');
    } catch (err) {
        console.error('Error:', err);
    } finally {
        await client.close();
    }
}

main().catch(console.error);

File: content/search/source/includes/string-tutorial/code-snippets/csharp/CreateMaterializedView.cs
using MongoDB.Bson;
using MongoDB.Driver;
using System;

public class CreateMaterializedView
{
    // Replace the placeholder with your Atlas connection string
    private const string uri = "<connectionString>";
    public static void Main(string[] args)
    {
        try
        {
            // Connect to your cluster
            var client = new MongoClient(uri);
            var database = client.GetDatabase("sample_airbnb");
            var collection = database.GetCollection<BsonDocument>("listingsAndReviews");

            // Defines the aggregation pipeline
            var pipeline = new BsonDocument[] 
            {
              new BsonDocument("$project", new BsonDocument
              {
                { "lastScrapedDate", new BsonDocument("$dateToString", new BsonDocument
                  {
                    { "format", "%Y-%m-%d" },
                    { "date", "$last_scraped" }
                  })
                },
                { "accommodatesNumber", new BsonDocument("$toString", "$accommodates") },
                { "maximumNumberOfNights", new BsonDocument("$toString", "$maximum_nights") },
                { "propertyName", "$name" },
                { "propertyType", "$property_type" }
              }),
              new BsonDocument("$merge", new BsonDocument
              { 
                { "into", "airbnb_mat_view" }, 
                { "whenMatched", "replace" } 
              })
            };

            // Executes the aggregation pipeline
            collection.Aggregate<BsonDocument>(pipeline);
            Console.WriteLine("Materialized view created!");
        }
        catch (Exception e)
        {
            Console.WriteLine($"Exception: {e.Message}");
        }
    }
}
